databend-jdbc/src/main/java/com/databend/jdbc/DatabendConnection.java

```diff
@@ -635,16 +675,35 @@ DatabendClient startQueryWithFailover(String sql, StageAttachment attach) throws
                 throw new SQLException("Error start query: " + "SQL: " + sql + " " + e.getMessage() + " cause: " + e.getCause(), e);
             }
             try {
+                // route hint is used when transaction occured,
+                // transaction procedure:
+                // 1. server return session body where txn state is active
+                // 2. when there is a active transaction, it will route all query to target route hint uri if exists
+                // 3. if there is not active transaction, it will use load balancing policy to choose a host to execute query
+                String query_id = UUID.randomUUID().toString();
+                String candidateHost = this.driverUri.getUri(query_id).toString();
```

Comment:
i guess it can be replaced like:

```
   String candidateHost = this.driverUri.getUri(query_id, routeHint).toString();
```

we could regard the route hint as one useful parameter for the load balancer.

Reply:
routeHint might not nessecary to contain the full address in the future, one possible way is it's a hash.

instead of unpack the encoded uri in the routehint, the discoverer can also consider making a simple loop to check which endpoint matches the hash.